target/consensource-api
# Update factory query to use fts for addresses

## Proposed change/fix

Add address querying support with postgres full text search, see the [database crate](https://github.com/target/consensource-database/pull/11) for more info.

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_


## How to run/test

1. Clone down the [consensource-compose](https://github.com/target/consensource-compose) repo:
```bash
 git clone --recurse-submodules https://github.com/target/consensource-compose.git
```
2. Make sure each submodule is up-to-date with the latest code changes
```sh
 git submodule update --remote $api
```
3. `cd api` and checkout this branch for api submodule
4. Build a local image for this image
```sh
 ./docker-helper.sh --build api
```
5. Build local images for other submodules as well if needed (`./docker-helper.sh --build local`)
6. Run `docker-compose -f docker-compose.run-local.yaml up`
7. Exec into the CLI container
8. Create an agent
```bash
 csrc agent create cliagent --url http://api:9009
```
9. Create an ingestion organization
```bash
 csrc organization create asserterorg 4 assertercontactname 7632918628 ENG --city minneapolis --country usa --street_address '821 thor street' --url http://api:9009
```
10. Create a factory assertion
```
 csrc assertion factory create {org_id_here} testfactoryassertion1 yuki 6123982765 JPN --city osaka --country japan --street_address 'fukuoka south st' --state_province kanto --postal_code 55414 --url http://api:9009
```
```
csrc assertion factory create {org_id_here} testfactoryassertion2 ren 6123982765 CHN --city shenzhen --country china --street_address 'dong east st' --state_province guangdong --postal_code 55446 --url http://api:9009
```
11. Query the API with one of the newly implemented parameters.
```
 http://localhost:9009/api/factories?address=shenzhen
```
